### PR TITLE
Add missing type hints - Part 1

### PR DESCRIPTION
**Pull request by finswimmer**
This is the first part of incremental PR with the goal to force type hints in application code and tests.

This PR adds missing typehints to:

* application code
* tests.config
* test.console

The `conftest.py` in the tests root folder will be updated whenever needed.

### CODE REVIEW

**Review by neersighted — CHANGES_REQUESTED** [review]
LGTM!

One question, besides the whitespace nit-picking:
Can/should we turn on Mypy for the affected modules, now that they are fully hinted? As best I can tell, nothing is actually checking that these new hints are correct at the moment.

File: pyproject.toml [hunk 1]
```diff
@@ -64,6 +64,7 @@ pytest-sugar = "^0.9"
 httpretty = "^1.0"
 zipp = { version = "^3.4", python = "<3.8"}
 deepdiff = "^5.0"
+typing-extensions = {version = "^4.0.0", python = "<3.8"}
```
[the remark this answers is not in the document]
Comment (finswimmer) [reply pyproject.toml:67]: Because this is how `poetry add` inserted it, I would prefer leaving it like it is.

**Comment by finswimmer** [comment]
> One question, besides the whitespace nit-picking: Can/should we turn on Mypy for the affected modules, now that they are fully hinted? As best I can tell, nothing is actually checking that these new hints are correct at the moment.

I let mypy inspect these modules. Some of the errors seems to be due to incorrect typings in the application code. This is why I suggest we first fix mypy errors there and have a look at the tests afterwards.

It looks like there are also some tricky errors, e.g.

```
tests/console/commands/test_show.py:1524: error: "Locker" has no attribute "mock_lock_data"; maybe "_lock_data" or "lock_data"?
```

This is because the `Poetry` object gets a `TestLocker` injected, which inherits from `Locker` but has some additional methods. Not sure how to teach this to mypy ...

TL;DR: Let us postpone mypy checks for the tests modules.

**Review by neersighted — APPROVED** [review]
